### PR TITLE
Source Files Reorganization

### PR DESCRIPTION
**Pull request by bcaddy**
- Moved all the source files in `src` to appropriate subdirectories. This included making several new directories. The only remaining file in `src` is `main.cpp`
- Deleted the `.gitignore` and `src.udb` files within `src`
- Deleted spare copy of `cloud_coolingcurve.text` that was in `src`. The primary copy is in `src/cooling`

### CODE REVIEW

File: src/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-*.o
```
Comment (twhite-cray) [inline src/.gitignore:1]: This file hides all the `.o` files from a build when running `git` commands. Without it, won't all the `.o` files clog up `git status` and `git commit` commands? We do not want to accidentally commit `.o` files to this repo. Why was the file removed?

Comment (twhite-cray) [reply src/.gitignore:1]: Oops. Looks like there's a top-level `.gitignore` that hides `.o`, and this one is redundant. Right?

**Comment by bcaddy** [comment]
That's correct. The top level .gitignore​ takes care of .o files so I removed this extranous one.

________________________________

**Review by bcaddy — COMMENTED** [review]
That's correct.

**Pull request closed without merging**